### PR TITLE
[enterprise-4.14] OSDOCS-5051: update RN for haproxy

### PR DESCRIPTION
**Pull request by jldohmann**
Version(s): 4.14

Issue: https://issues.redhat.com/browse/OSDOCS-5051

Link to docs preview: https://63077--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-14-release-notes#ocp-4-14-networking-haproxy-update

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by jldohmann** [comment]
@ShudiLi PTAL for QE ack. This PR adds the HAProxy version bump to the OCP 4.14 release notes. Thank you!

File: release_notes/ocp-4-14-release-notes.adoc [hunk 1]
```diff
@@ -144,6 +144,10 @@ For example, in some scenarios, it is a priority to maximize CPU and memory reso
 
 For more information about this more flexible SR-IOV network deployment during NUMA-aware pod scheduling, see xref:../networking/hardware_networks/configuring-sriov-device.adoc#nw-sriov-exclude-topology-manager_configuring-sriov-device[Exclude the SR-IOV network topology for NUMA-aware scheduling].
 
+[id="ocp-4-14-networking-haproxy-update"]
+==== Update to HAProxy 2.6
+{product-title} updated to HAProxy 2.6.
```
Comment (mburke5678) [inline release_notes/ocp-4-14-release-notes.adoc:149]: > With this release, {product-title} updated to HAProxy 2.6.

Added _With this release_ to match most others and I think you need the verb here.

**Comment by ShudiLi** [comment]
/lgtm
thanks

File: release_notes/ocp-4-14-release-notes.adoc (hunk 1 above)
Comment (jldohmann) [reply release_notes/ocp-4-14-release-notes.adoc]: thanks! updated

**Pull request merged**